src/client/Microsoft.Identity.Client/WsTrust/MexDocument.cs

```diff
@@ -136,6 +150,20 @@ private void ReadPolicies(XContainer mexDocument)
         private void ReadPolicyBindings(XContainer mexDocument)
         {
             IEnumerable<XElement> bindingElements = mexDocument.Elements().First().Elements(XmlNamespace.Wsdl + "binding");
+
+            //Unable to find binding information in root node. Attempting to search document for binding information
+            if (bindingElements.Count() == 0)
```

Comment:
This seems very similar code, with a different parent to the collection (mexDocument, binding) and different value to search ("Policy", "binding", "service")
you might want to extract this code into a method ?

Reply:
Sorry, this wasn't the right version of the branch. I was in a hurry yesterday and forgot to commit my latest refactoring.

Thanks for looking @jmprieur